.github/workflows/actions.yml

```diff
@@ -70,8 +70,10 @@ jobs:
         run: |
           pip install -r requirements.txt --progress-bar off
           pip install --no-deps -e "." --progress-bar off
-      - name: Lint
-        run: bash shell/lint.sh
+      - name: Install pre-commit
+        run: pip install pre-commit
+      - name: Run pre-commit
+        run: pre-commit run --all-files
```

Comment:
There is a difference between `format` and `lint` that got lost here.

`format` fixes the files. `lint` errors if anything needs to be changed.

In a PR, it's too late to fix the files, but we need to detect incorrectly formatted files and throw an error. Is it possible to do that?

`lint` runs each command and fails if any command fails. `format` runs all of them regardless and doesn't fail.

Reply:
`format` in pre-commit will fail if the code is non-compliant and if it has to format the file.